### ISSUE TITLE
[Enhancement][opensearch][opensearch-dashboards] Lifecycle Probes

### ISSUE THREAD

**Issue by hgoscenski-imanage** [opensearch-project/opensearch-devops#44]
**Is your feature request related to a problem? Please describe.**
I think there ought to be liveness/readiness/startup probes for the opensearch statefulset and the opensearch-dashboards deployment.

Currently the pods are immediately marked ready on startup, leading to more than one of the nodes being unavailable which can lead to instability especially with lots of data.

**Describe the solution you'd like**
I am happy to implement something in a PR to resolve this, if that is something the project would want and my general ideas below sound good.

Opensearch:
Readiness: query health endpoint, which may require whitelisting or auth, could potentially look for green cluster status
Liveness: can check for tcp connection to 9200 or 9300
Startup: can check for health endpoint regardless of status?

Opensearch-dashboards:
Readiness: Health is at least yellow in /api/status
Liveness: 5601 is open to tcp
Startup: same as readiness

**Describe alternatives you've considered**
NA

**Additional context**
I am happy to implement if the ideas sound good.

**Comment by TheAlgo** [comment]
I completely agree to this and it is kind of compulsory. I also wanted to have something like [this](https://github.com/opensearch-project/opensearch-devops/blob/main/Helm/opensearch/values.yaml#L314) included as readiness as well.

**Comment by hgoscenski-imanage** [comment]
That would be clever, do we want to include a sane default, or would we want to leave it disabled until the user chooses to enable it?

**Comment by TheAlgo** [comment]
> That would be clever, do we want to include a sane default, or would we want to leave it disabled until the user chooses to enable it?

What is we can have options to choose between different probes. As in have a basic one and some advanced one as well. Just thinking to make the chart as robust as possible.
